app/src/main/java/com/zeapo/pwdstore/git/GitServerConfigActivity.kt

```diff
@@ -71,42 +71,45 @@ class GitServerConfigActivity : BaseGitActivity() {
             }
         }
 
-        binding.serverUrl.setText(GitSettings.url)
+        binding.serverUrl.setText(GitSettings.url.also {
+            if (it.isNullOrEmpty()) return@also
+            setAuthModes(it.startsWith("http://") || it.startsWith("https://"))
+        })
         binding.serverBranch.setText(GitSettings.branch)
 
         binding.serverUrl.doOnTextChanged { text, _, _, _ ->
             if (text.isNullOrEmpty()) return@doOnTextChanged
-            if (text.startsWith("http://") || text.startsWith("https://")) {
-                binding.authModeSshKey.isVisible = false
-                binding.authModeOpenKeychain.isVisible = false
-                binding.authModePassword.isVisible = true
-                if (binding.authModeGroup.checkedButtonId != binding.authModePassword.id)
-                    binding.authModeGroup.check(View.NO_ID)
-            } else {
-                binding.authModeSshKey.isVisible = true
-                binding.authModeOpenKeychain.isVisible = true
-                binding.authModePassword.isVisible = true
-                if (binding.authModeGroup.checkedButtonId == View.NO_ID)
-                    binding.authModeGroup.check(binding.authModeSshKey.id)
-            }
+            setAuthModes(text.startsWith("http://") || text.startsWith("https://"))
         }
 
         binding.saveButton.setOnClickListener {
             val newUrl = binding.serverUrl.text.toString().trim()
             // If url is of type john_doe@example.org:12435/path/to/repo, then not adding `ssh://`
             // in the beginning will cause the port to be seen as part of the path. Let users know
             // about it and offer a quickfix.
```

Comment:
We should probably update this comment.

Reply:
Will add a note to do it after the v1.13.0 release.